File: packages/el/etest/ios/CompatibleNitroView.m

#ifndef RCT_NEW_ARCH_ENABLED
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>
#if __has_include(<etest/etest-Swift.h>)
#import <etest/etest-Swift.h>
#else
#import "etest-Swift.h"
#endif

@interface CompatibleNitroViewManager : RCTViewManager
@end

@implementation CompatibleNitroViewManager

RCT_EXPORT_MODULE()

#if __has_include(<UIKit/UIKit.h>)
- (UIView *)view {
  return [[LegacyCompatibleNitroView alloc] init];
}
#elif __has_include(<AppKit/AppKit.h>)
- (NSView *)view {
  return [[LegacyCompatibleNitroView alloc] init];
}
#endif

RCT_EXPORT_VIEW_PROPERTY(color, NSString)
RCT_EXPORT_VIEW_PROPERTY(startFrom, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onTick, RCTDirectEventBlock)

RCT_EXPORT_METHOD(reset:(nonnull NSNumber *)reactTag)
{
  [self.bridge.uiManager addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, NSView *> *viewRegistry) {
    NSView *view = viewRegistry[reactTag];
    if ([view isKindOfClass:[LegacyCompatibleNitroView class]]) {
      [(LegacyCompatibleNitroView *)view reset];
    }
  }];
}
@end

#endif

File: packages/el/etest/ios/CompatibleNitroView.swift
//#if  RCT_NEW_ARCH_ENABLED
 //test
//#endif

#if canImport(UIKit)
class HybridCompatibleNitroView : HybridCompatibleNitroViewSpec {


  // UIView
  var view: UIView = UIView()

  // props
  var color: String = "#000" {
    didSet {
      view.backgroundColor = hexStringToUIColor(hexColor: color)
    }
  }

  func hexStringToUIColor(hexColor: String) -> UIColor {
    let stringScanner = Scanner(string: hexColor)

    if(hexColor.hasPrefix("#")) {
      stringScanner.scanLocation = 1
    }
    var color: UInt32 = 0
    stringScanner.scanHexInt32(&color)

    let r = CGFloat(Int(color >> 16) & 0x000000FF)
    let g = CGFloat(Int(color >> 8) & 0x000000FF)
    let b = CGFloat(Int(color) & 0x000000FF)

    return UIColor(red: r / 255.0, green: g / 255.0, blue: b / 255.0, alpha: 1)
  }
}
#elseif canImport(AppKit)
class HybridCompatibleNitroView : HybridCompatibleNitroViewSpec {
  lazy var view: NSView = CompatibleNitroViewShare()

  override init() {
    super.init()
  }

  // props
  var color: String = "#000" {
    didSet {
    #if SWIFT_CXX_INTEROP
    print("Using modern C++ interop")
    #else
    print("Using legacy ObjC bridge")
    #endif
      (view as? CompatibleNitroViewShare)?.color = color
    }
  }
  var startFrom: Double? {
    didSet {
        (view as? CompatibleNitroViewShare)?.startFrom = NSNumber(value: startFrom ?? 0)
    }
  }
  var onTick: ((Double) -> Void)? {
      didSet {
          (view as? CompatibleNitroViewShare)?.onTick = { [weak self] data in
              if let count = data["count"] as? Double {
                  self?.onTick?(count)
              }
          }
      }
  }

  func reset() throws {
      (view as? CompatibleNitroViewShare)?.reset()
  }
}
#endif
